src/main/java/edu/iu/uits/mail/MailSigner.java

```diff
@@ -78,12 +79,20 @@ public MailSigner(Properties properties) {
             }).findFirst();
 
             if(signingAddress.isPresent()) {
-                String emailKeyPassword = properties.getProperty(String.format("mail.keystore.%s.password", signingAddress.get()));
-                if(emailKeyPassword == null || emailKeyPassword.trim().isEmpty()) {
-                    log.debug(String.format("No key password found for %s.  Defaulting to using the keystore password.", signingAddress.get()));
-                    emailKeyPassword = properties.getProperty("mail.keystore.password");
+                String alias = signingAddress.get();
+                // check for full alias (signing address) in properties
+                String emailKeyPassword = properties.getProperty(String.format("mail.keystore.%s.password", alias));
+                if (emailKeyPassword == null || emailKeyPassword.trim().isEmpty()) {
+                    // try local part of address
+                    log.debug(String.format("No key password found for %s.  Trying local portion, %s.", alias, alias.replaceAll(localAddressRegex, "$1")));
+                    emailKeyPassword = properties.getProperty(String.format("mail.keystore.%s.password", alias.replaceAll(localAddressRegex, "$1")));
+                    if (emailKeyPassword == null || emailKeyPassword.trim().isEmpty()) {
+                        // default to keystore password
+                        log.debug(String.format("No key password found for %s.  Defaulting to using the keystore password.", alias));
+                        emailKeyPassword = properties.getProperty("mail.keystore.password");
+                    }
```

Comment:
This all looks fine but is a lot of nested if statements.  Some unit tests to cover this logic would be great.